premarum-web-client/components/departmentChoiceModal.tsx

```diff
@@ -0,0 +1,150 @@
+                                    <CloseRounded sx={classes.closeButtonIcon}/>
+                                </IconButton>
+                            :<></>}
+                        </Grid>
+
+                        <Divider/>
+
+                        <CardContent sx={classes.cardContent}>
+                            {modalLoading ? (
+                                <Stack alignItems="center" justifyContent="center">
+                                    <CircularProgress/>
+                                </Stack>
+                            ) : (
+                                <>
+                                    <Autocomplete
+                                        id="tags-outlined"
+                                        options={props.departments}
+                                        getOptionLabel={(option) => `${option.departmentName}`}
+                                        filterSelectedOptions
+                                        defaultValue={props.departments.find(b => b.departmentId === student?.departmentId)}
+                                        onChange={(event, newValue) => {handleDepartmentChange(newValue?.departmentId)}}
+                                        renderInput={(params) => (
+                                            <TextField
+                                                {...params}
+                                                label="Department Selection"
+                                                placeholder="Select a Department..."
+                                            />
+                                        )}
+                                    />
+                                </>
+                            )}
+                        </CardContent>
+
+                        <Divider/>
+
+                        <CardActions sx={classes.cardActions}>
+                            <Button
+                                disabled={modalLoading || !studentDept}
+                                onClick={handleUpdateDepartment}
+                            >
+                                    Submit
+                            </Button>
+                        </CardActions>
+                    </Card>
+                </Grid>
+            </Fade>
+        </Modal>
+    )
+}
+
+const useStyles = {
+    modalGridMain: {
+        width: '100%',
+        height: '100%',
+    },
+    modalCard: {
+        width: '60%',
+        minHeight: '30%',
+        backgroundColor: grey[50],
+    },
+    closeButton: {
+        marginRight: 1,
+    },
+    closeButtonIcon: {
+        fontSize: '30px',
+    },
+    cardContent: {
+        width: '100%',
+        height: '100%',
+        padding: '60px 20px',
+    },
+    titleInput: {
+        marginBottom: 1,
+    },
+    cardActions: {
+        justifyContent: 'flex-end',
+        alignItems: 'flex-end',
+    },
+};
+
+const classes = useStyles;
+
+export async function getStaticProps() {
```

Comment:
This getStaticProps wont work because getStaticProps Only works in page components like Home